.github/workflows/win_ci.yml

```diff
@@ -0,0 +1,260 @@
+
+    # The name of the test follow the tested language
+    name: >
+      ${{ matrix.SWIGLANG }} ${{ matrix.VER }}
+      ${{ matrix.COMPILER || 'msvc' }} ${{ matrix.os }}
+      ${{ matrix.continue-on-error && '(can fail)' }}
+
+    strategy:
+      matrix:
+        include:
+        - SWIGLANG: csharp
+          INSTALL: 'true'
+        - SWIGLANG: csharp
+          os: 'windows-2019'
+          INSTALL: 'true'
+        - SWIGLANG: java
+          VER: 8
+        - SWIGLANG: java
+          VER: 11
+        - SWIGLANG: java
+          VER: 17
+       #  TODO JDK 21 fails on testing
+       #- SWIGLANG: java
+       #  VER: 21
+        - SWIGLANG: python
+          VER: '3.7'
+        - SWIGLANG: python
+          VER: '3.8'
+        - SWIGLANG: python
+        # VER: '3.9'
+        - SWIGLANG: python
+          VER: '3.10'
+        - SWIGLANG: python
+          VER: '3.11'
+        - SWIGLANG: python
+          VER: '3.12'
+        - SWIGLANG: python
+          INSTALL: 'true'
+          COMPILER: gcc
+      # Run all of them, as opposed to aborting when one fails
+      fail-fast: false
+
+    env:
+      CFLAGS: '-O2'
+      CXXFLAGS: '-O2'
+      CCCL_OPTIONS: '--cccl-muffle /W3 /EHsc'
+      PCRE2_CCCL_LD: '-lpcre2-8-static --cccl-link /NODEFAULTLIB:MSVCRT'
+      CHECK_OPTIONS: 'CSHARPOPTIONS=-platform:x64'
+      SWIGLANG: ${{ matrix.SWIGLANG }}
+      COMPILER: ${{ matrix.COMPILER }}
+      INSTALL: ${{ matrix.INSTALL }}
+      VER: ${{ matrix.VER }}
+      OS: ${{ matrix.os }}
+
+# cl.exe:
+#  https://learn.microsoft.com//cpp/build/reference/compiler-options
+#  /EHc  extern "C" defaults to nothrow.
+#  /EHs  Enable C++ exception handling (no SEH exceptions).
+#  /W3   Warning level.
+#  /WX   Treat warnings as errors.
+#  https://learn.microsoft.com//cpp/build/reference/linker-options
+#  cl.exe pass linker:
+#  /VERBOSE:LIB        Outputs progress messages during the link process.
+#  /NODEFAULTLIB:lib   Ignore library 'lib'
+
+    steps:
+    - name: Machine Info
+      shell: cmd
+      run: |
+          systeminfo | findstr /B /C:"OS Name" /B /C:"OS Version"
+
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        show-progress: true
+
+    - name: Install CCache
+      uses: hendrikmuhs/ccache-action@v1
+      with:
+        key: ${{ matrix.os || 'windows-2022' }}-${{ matrix.COMPILER || 'msvc' }}
+
+    - name: Install NuGet Packages
+      if: ${{ env.COMPILER == '' }}
+      shell: cmd
+      run: |
+          rem 'nuget build for .NET: https://www.nuget.org/packages'
+          nuget install PCRE2 -OutputDirectory C:\Tools
+          nuget install boost -OutputDirectory C:\Tools
+
+    - name: Setup MSVC
+      if: ${{ env.COMPILER == '' || env.SWIGLANG == 'csharp' }}
+      uses: TheMrMilchmann/setup-msvc-dev@v3
+      with:
+        arch: x64
+
+    - name: Prepare Environment
+      shell: bash
+      run: |
+          uname --all
+          # Tabs are used for here-documents indentation!
```

Comment:
Is it not possible to use spaces everywhere including the here-documents? Tabs are probably best avoided as no-one has ever got the same settings. We use 8 spaces for tabs generally in SWIG and it looks like you're using 4.

Reply:
Unfortunately bash here-documents with "<<-" permit tabs only, while YAML  uses spaces only.
The only solution is to skip here-documents, 
and use multiple `echo` with pipe.

Reply:
Yuck.
Yes, maybe multiple echo statements will work, there is already one like it so just use this approach:
```
            echo "JAVA_HOME=$java_path" >> $GITHUB_ENV
```

Reply:
Done